File: solutions/erlang/grains/1/grains.erl
-module(grains).
-export([square/1, total/0]).

square(Position) ->
  square(Position, 1).

square(1, Acc) ->
  Acc;
square(Position, Acc) ->
  square(Position - 1, Acc * 2).

total() ->
  lists:sum([square(G) || G <- lists:seq(1, 64)]).

File: solutions/lua/grains/1/grains.lua
local grains = {}

function grains.square(n)
	if n == 64 then
		return 2.0 * (1 << n - 2)
	else
		return 1 << n - 1
	end
end

function grains.total()
	local sum = 0

	for n = 1, 64 do
		sum = sum + grains.square(n)
	end

	return sum
end

return grains
